src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -1281,6 +1304,186 @@ internal BoundExpression BindCollectionBuilderCreate(
+                                containingType,
+                                methodName,
+                                parameters: [("comparer", comparerType)],
+                                returnType,
+                                diagnostics);
+                            if ((object)targetType.OriginalDefinition == Compilation.GetWellKnownType(WellKnownType.System_Collections_Generic_IDictionary_KV))
+                            {
+                                var intType = TypeWithAnnotations.Create(GetSpecialType(SpecialType.System_Int32, diagnostics, syntax));
+                                addSignature(
+                                    builder,
+                                    Compilation,
+                                    syntax,
+                                    WellKnownMember.System_Collections_Generic_Dictionary_KV__ctor_Int32,
+                                    containingType,
+                                    methodName,
+                                    parameters: [("capacity", intType)],
+                                    returnType,
+                                    diagnostics);
+                                addSignature(
+                                    builder,
+                                    Compilation,
+                                    syntax,
+                                    WellKnownMember.System_Collections_Generic_Dictionary_KV__ctor_Int32_IEqualityComparer_K,
+                                    containingType,
+                                    methodName,
+                                    parameters: [("capacity", intType), ("comparer", comparerType)],
+                                    returnType,
+                                    diagnostics);
+                            }
+                        }
+                        break;
+                    default:
+                        throw ExceptionUtilities.UnexpectedValue(collectionTypeKind);
+                }
+            }
+
+            static void addSignature(
+                ArrayBuilder<MethodSymbol> builder,
+                CSharpCompilation compilation,
+                SyntaxNode syntax,
+                WellKnownMember wellKnownConstructor,
+                NamedTypeSymbol containingType,
+                string methodName,
+                ImmutableArray<(string Name, TypeWithAnnotations Type)> parameters,
+                TypeWithAnnotations returnType,
+                BindingDiagnosticBag diagnostics)
+            {
+                var constructor = (MethodSymbol?)GetWellKnownTypeMember(compilation, wellKnownConstructor, diagnostics, syntax: syntax);
+                if (constructor is { })
+                {
+                    var method = new CollectionArgumentsSignatureOnlyMethodSymbol(constructor, methodName, containingType, parameters, returnType);
```

Comment:
Should the return type be `void` or come from `constructor`, instead of be the target/collection type?

Reply:
Lowering is responsible for rewriting the bound arguments and signature to a constructor call or equivalent initialization. For initial binding, it seemed sufficient to bind to a well-defined signature only.